packages/compass/src/main/application.ts

```diff
@@ -301,7 +301,41 @@ class CompassApplication {
       try {
         const proxyOptions = proxyPreferenceToProxyOptions(value);
         await app.whenReady();
-        await target.setProxy(translateToElectronProxyConfig(proxyOptions));
+
+        try {
+          const electronProxyConfig =
+            translateToElectronProxyConfig(proxyOptions);
+          await target.setProxy(electronProxyConfig);
+        } catch (err) {
+          const headline = String(
+            err && typeof err === 'object' && 'message' in err
+              ? err.message
+              : err ||
+                  'Currently Compass does not support authenticated or ssh proxies.'
+          );
+
+          log.warn(
+            mongoLogId(1_001_000_328),
+            logContext,
+            'Unable to set proxy configuration',
+            {
+              error: headline,
+            }
+          );
+
+          const sep = path.sep;
+          const configPath = `${app.getPath(
+            'userData'
+          )}${sep}AppPreferences${sep}General.json`;
+
+          dialog.showErrorBox(
+            'Unsupported proxy configuration',
+            `${headline}\n\n
+            To reset the proxy configuration, remove the "proxy" key in ${configPath} and restart Compass.`
+          );
+
+          app.quit();
```

Comment:
Why do we quit here? Does the error popup also get closed when the app quits? Haven't tried.

Reply:
The application will continue running and crash because of the proxy missconfig, showing another error and another stack trace. Early quitting avoid this and gives the user all the context to action on the error.

Do you have any suggestion on how to improve it?

Reply:
Gotcha, just checking, no suggestion on improvement, I'm not too familiar with the issue. I was worried the app.quit would make the popup `showErrorBox` disappear as well.